test/unit/Dredd-test.js

```diff
@@ -1,759 +1,94 @@
-const bodyParser = require('body-parser');
-const express = require('express');
-const fsStub = require('fs');
-const proxyquire = require('proxyquire').noCallThru();
-const requestStub = require('request');
 const sinon = require('sinon');
-const path = require('path');
 const { assert } = require('chai');
-const parse = require('dredd-transactions/parse');
-const compile = require('dredd-transactions/compile');
 
-const loggerStub = require('../../lib/logger');
+const Dredd = require('../../lib/Dredd');
 
-const parseStub = sinon.spy(parse);
-const compileStub = sinon.spy(compile);
+const { _prefixErrors: prefixErrors, _toTransactions: toTransactions } = Dredd;
 
-const Dredd = proxyquire('../../lib/Dredd', {
-  request: requestStub,
-  'dredd-transactions/parse': parseStub,
-  'dredd-transactions/compile': compileStub,
-  fs: fsStub,
-  './logger': loggerStub,
-});
-
-
-function compareLocation(ad1, ad2) {
-  if (ad1.location < ad2.location) { return -1; }
-  if (ad1.location > ad2.location) { return 1; }
-  return 0;
-}
-
-
-describe('Dredd class', () => {
```

Comment:
I've deleted most of the tests, because I'm not sure they were useful at all. I'm not fan of unit tests based on stubbing and mocking everything around. I think such tests aren't very valuable. Unfortunately Dredd's test suite is full of such tests.

My strategy would be to push as much logic as possible into pure, nicely testable functions, which can get verified by unit tests. Then to test the rest, the glue, by integration tests. Therefore I'm adding to the integration tests rather than stubbing and mocking everything here.